lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/fs.rb

```diff
@@ -376,6 +420,11 @@ def cmd_ls(*args)
     order = args.include?('-r') ? :reverse : :forward
     args.delete('-r')
 
+    recursive = args.include?('-R')
+    args.delete('-R')
```

Comment:
Since Array#delete returns `nil` if the argument isn't found, this can be simplified to:
```ruby
recursive = args.delete('-R')
```